app/controllers/concerns/application_shared.rb

```diff
@@ -24,6 +24,7 @@ def current_permission
 
   def set_taxonomy
     user = User.current
+    TopbarSweeper.expire_cache
```

Comment:
I think this would be better to be moved one line above, it does not depend on user, `expire_cache` has it's own check for User.current being nil